### PR TITLE
refactor: a lot of stuff

### PR DESCRIPTION
**Pull request by elianiva**
- add file aggregator
- move out some constants
- avoid truthy/falsy values
- utilise path alias more often
- refactor some dodgy logic
- update /snap constant to be in sync with latest shiki
- more jsdoc
- convert Semaphore to a class-style syntax
- add some guards
- bump deps
- fix dukun (hopefully)

### CODE REVIEW

**Comment by elianiva** [comment]
muncul anomali, `JSON.stringify(eval("BigInt(1)"), (_, v) => typeof v)` hasilnya itu bisa jadi `string` anjay, bukan `bigint`
ga nyentuh anything related to eval juga, bump dependency cuma eslint-config

di repo ini:
<kbd>
image
</kbd>

di REPL nodejs:
<kbd>
image
</kbd>

di graphene:
<kbd>
image
</kbd>

ga ngerti lagi dah kenapa bisa gini

File: src/services/pastebin/index.js [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
 import { sizeInBytes } from "#utils/size.js";
 import { terminal } from "#utils/logger/terminal.js";
+import { logger } from "#utils/logger/index.js";
 
 export const PASTEBIN_FILE_TOO_BIG =
   "Can't create pastebin. Text is bigger than 5 MB";
 
 /**
- *
+ * makeRequest is a function to make a request to Polarite.
```
Comment (aldy505) [inline src/services/pastebin/index.js:11]: this reminds me because Polarite is also brroken right now. hadeh.

File: src/services/quiz/index.js [hunk 2]
```diff
@@ -68,64 +68,63 @@ async function quiz(context, mongo, cache) {
     return;
   }
 
-  const quizes = await Poll.find({ posted: false });
-  const pickQuiz = quizes[randomNumber(0, quizes.length - 1)];
+  const polls = await Poll.find({ posted: false });
```
Comment (aldy505) [inline src/services/quiz/index.js:71]: should add a check whether this polls is an empty array or not.

Comment (aldy505) [reply src/services/quiz/index.js:71]: if it is, meaning there are no available quiz left

**Review by aldy505 — COMMENTED** [review]
a few minor notes

File: src/services/pastebin/index.js (hunk 1 above)
Comment (elianiva) [reply src/services/pastebin/index.js:11]: WKWKWKW

**Pull request merged**